runtime/compiler/aarch64/codegen/J9TreeEvaluator.cpp

```diff
@@ -5028,7 +5038,13 @@ static TR::Register *VMinlineCompareAndSwapObject(TR::Node *node, TR::CodeGenera
    fourthChild = node->getChild(3);
    fifthChild = node->getChild(4);
 
-   objReg = cg->evaluate(secondChild);
+#if defined(J9VM_GC_ENABLE_SPARSE_HEAP_ALLOCATION)
+   bool isObjOffHeapDataAddr = (gcMode == gc_modron_wrtbar_cardmark_incremental && TR::Compiler->om.isOffHeapAllocationEnabled() && secondChild->isDataAddrPointer());
+   TR::Register *baseObjReg = NULL;
+   if (isObjOffHeapDataAddr)
+      TR::TreeEvaluator::stopUsingCopyReg(secondChild->getFirstChild(), baseObjReg, cg);
```

Comment:
is it possible that optimizer transformed the arrayCopy tree to such an extent that secondChild doesn't have children anymore (for example, it is an aRegLoad or something like that)? two obvious problems:  1) is it still having isDataAddrPointer flag set?  2) getFirstChild will have or cause problems.

This comment is applicable to all files modified.

Reply:
Given that card marking code is ready, I moved the assert code to a different PR to merge the card marking code.
https://github.com/eclipse-openj9/openj9/pull/20480